core/src/main/scala/chisel3/internal/firrtl/Converter.scala

```diff
@@ -302,7 +302,15 @@ private[chisel3] object Converter {
 
   def extractType(data: Data, info: SourceInfo): fir.Type = extractType(data, false, info)
 
-  def extractType(data: Data, clearDir: Boolean, info: SourceInfo): fir.Type = data match {
+  def extractType(data: Data, clearDir: Boolean, info: SourceInfo, checkConst: Boolean = true): fir.Type = {
+    if (checkConst && data.isConst) {
+      fir.ConstType(extractType(data, clearDir, info, false))
+    } else {
+      extractTypeImpl(data, clearDir, info)
+    }
+  }
+
+  def extractTypeImpl(data: Data, clearDir: Boolean, info: SourceInfo): fir.Type = data match {
```

Comment:
Can this be merged with the `extractTypeImpl` as opposed to having a new method?

Something like:

```
  case _ if checkConst && data.isConst => 
  case _: Clock => 
  ...
```

Reply:
This looks much cleaner, thanks!